webpack.config.js

```diff
@@ -1,71 +1,78 @@
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const webpack = require('webpack');
+const path = require('path');
+var dotenv = require('dotenv').config({ path: __dirname + '/.env' });
 
-const urlDev = "https://localhost:3000/";
-const urlProd = "https://www.contoso.com/"; // CHANGE THIS TO YOUR PRODUCTION DEPLOYMENT LOCATION
+const urlDev = 'https://localhost:3000/';
+const urlProd = 'https://www.contoso.com/'; // CHANGE THIS TO YOUR PRODUCTION DEPLOYMENT LOCATION
 
 module.exports = async (env, options) => {
-  const dev = options.mode === "development";
-  const buildType = dev ? "dev" : "prod";
+  const dev = options.mode === 'development';
+  const buildType = dev ? 'dev' : 'prod';
   const config = {
-    devtool: "source-map",
+    devtool: 'source-map',
     entry: {
-      polyfill: ["core-js/stable", "regenerator-runtime/runtime"],
-      vendor: ["react", "react-dom", "core-js", "@fluentui/react"],
-      taskpane: ["react-hot-loader/patch", "./src/taskpane/index.tsx"],
-      commands: "./src/commands/commands.ts",
+      polyfill: ['core-js/stable', 'regenerator-runtime/runtime'],
+      vendor: ['react', 'react-dom', 'core-js', '@fluentui/react'],
+      taskpane: ['react-hot-loader/patch', './src/taskpane/index.tsx'],
+      commands: './src/commands/commands.ts',
+    },
+    output: {
+      path: path.resolve(__dirname, 'dist'),
     },
     resolve: {
-      extensions: [".ts", ".tsx", ".html", ".js"],
+      extensions: ['.ts', '.tsx', '.html', '.js'],
+      fallback: {
+        Promise: require.resolve('es6-promise'),
+      },
     },
     module: {
       rules: [
         {
           test: /\.worker\.ts$/,
-          loader: "worker-loader",
+          loader: 'worker-loader',
         },
         {
           test: /\.tsx?$/,
-          use: ["react-hot-loader/webpack", "ts-loader"],
+          use: ['react-hot-loader/webpack', 'ts-loader'],
           exclude: /node_modules/,
         },
         {
           test: /\.css$/,
-          use: ["style-loader", "css-loader"],
+          use: [MiniCssExtractPlugin.loader, 'css-loader'],
         },
         {
           test: /\.(png|jpg|jpeg|gif)$/,
-          loader: "file-loader",
-          options: {
-            name: "[path][name].[ext]",
+          type: 'asset/resource',
+          generator: {
+            filename: '[path][name].[ext]',
           },
         },
       ],
     },
     plugins: [
       new CleanWebpackPlugin(),
       new webpack.DefinePlugin({
-        "process.env": JSON.stringify(dotenv.parsed),
+        'process.env': JSON.stringify(dotenv.parsed),
       }),
       new CopyWebpackPlugin({
         patterns: [
           {
-            to: "taskpane.css",
-            from: "./src/taskpane/taskpane.css",
+            to: 'taskpane.css',
+            from: './src/taskpane/taskpane.css',
           },
           {
-            to: "[name]." + buildType + ".[ext]",
```

Comment:
Shouldn't this be `[name][ext]`? At least the `start:web` command tries to open the `manifest.xml` file (and not something with the buildType). It's also used here https://github.com/OfficeDev/Office-Addin-TaskPane/blob/master/webpack.config.js#L75

Reply:
right